code/datums/elements/weapon_description.dm

```diff
@@ -0,0 +1,186 @@
+
+	// Meaningless flavor text. The number of crimes is constantly changing because of the complex Nanotrasen legal system and the esoteric nature of time itself!
+	readout += "<span class='warning'>WARNING:</span> This item has been marked as dangerous by the NT legal team because of its use in <span class='warning'>[rand(2,99)] [crimes[rand(1, crimes.len)]]</span> in the past hour.\n"
+
+	// Doesn't show the base notes for items that have the override notes variable set to true
+	if(!source.override_notes)
+		// Make sure not to divide by 0 on accident
+		if(source.force > 0)
+			readout += "Our extensive research has shown that it takes a mere <span class='warning'>[round((100 / source.force), 0.1)]</span> hits to down [victims[rand(1, victims.len)]] with no armor."
+		else
+			readout += "Our extensive research found that you couldn't beat yourself to death with this if you tried."
+
+		if(source.throwforce > 0)
+			readout += "If you decide to throw this object instead, one will take <span class='warning'>[round((100 / source.throwforce), 0.1)]</span> hits before collapsing."
+		else
+			readout += "If you decide to throw this object instead, then you will have trouble damaging anything."
+		if(source.armour_penetration > 0 || source.block_chance > 0)
+			readout += "This item has proven itself <span class='warning'>[weapon_tag_convert(source.armour_penetration)]</span> of piercing armor and <span class='warning'>[weapon_tag_convert(source.block_chance)]</span> of blocking attacks."
+	// Custom manual notes
+	if(source.offensive_notes)
+		readout += source.offensive_notes
+
+	// Check if we have an additional proc, if so, add it to the readout
+	if(attached_proc)
+		readout += call(attached_proc)(source)
+
+	// Finally bringing the fields together
+	return readout.Join("\n")
+
+/**
+ *
+ * Converts percentile based stats to an adjective appropriate for the
+ * examined warning label
+ *
+ * Arguments:
+ *  * tag_val: The value of the item to be added to the tag
+ */
+/datum/element/weapon_description/proc/weapon_tag_convert(tag_val)
+	switch(tag_val)
+		if(0)
+			return "INCAPABLE"
+		if(1 to 25)
+			return "BARELY CAPABLE"
+		if(26 to 50)
+			return "CAPABLE"
+		if(51 to 75)
+			return "VERY CAPABLE"
+		if(76 to INFINITY)
+			return "EXTREMELY CAPABLE"
+		else
+			return "STRANGELY CAPABLE"
+
+
+/**
+ *
+ * Outputs type-specific weapon stats for ballistic weaponry.
+ * It contains extra breaks for the sake of presentation
+ *
+ * Arguments:
+ * 	* source - The object being evaluated
+ */
+/datum/element/weapon_description/proc/add_notes_ballistic(obj/item/gun/ballistic/source)
+	if(source.magazine) // Make sure you have a magazine, thats where the warning is!
+		return "\nBe especially careful around this device, as it can be loaded with <span class='warning'>[source.magazine.caliber]</span> rounds, which you can inspect for more information."
```

Comment:
```suggestion
		return "\nBe especially careful around this device, as it can be loaded with <span class='warning'>[source.magazine.caliber]</span> round\s, which you can inspect for more information."
```